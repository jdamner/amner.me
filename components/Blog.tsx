import React from "react";
/* Layout */
import PostList from "./PostList";
/* Types */
import ContactDetails from "./ContactDetails";

export default function Blog({
  posts,
}: {
  posts: { slug: string; title: string }[];
}) {
  return (
    <div className="lg:grid lg:grid-cols-3">
      <PostList posts={posts} className="mt-5 col-span-1" />
      <ContactDetails />
    </div>
  );
}
