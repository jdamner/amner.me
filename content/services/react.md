---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
