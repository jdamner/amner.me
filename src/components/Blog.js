import React from "react";
import Title from "./Global/Title";

import PostList from "./PostList/PostList";


export default function Blog( { posts }) {
  return (
    <section className='blog' id='blog'>
      <div className='blog-wrap'>
        <Title text='My Projects and Writing' />
        <div className='blog-wrap-grid'>
          <PostList posts={posts} />
        </div>
        
      </div>
    </section>
  );
}