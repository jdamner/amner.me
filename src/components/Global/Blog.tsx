import React from "react";
/* Layout */
import Title from "./Title";
import PostList from "./PostList";
/* Types */
import type { PostLinkType } from "../../types";

export default function Blog({ posts }: { posts: PostLinkType[] }) {
  return (
    <>
      <Title title="Projects and Writing">Blog</Title>
      <PostList posts={posts} />
    </>
  );
}
