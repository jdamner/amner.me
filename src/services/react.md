---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.