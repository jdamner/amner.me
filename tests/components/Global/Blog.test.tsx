import { render } from "@testing-library/react"
import Blog from "../../../src/components/Global/Blog"
import { getTestPosts } from "../../helper"

describe("Blog", () => {
  it("renders correctly", () => {
    const { container } = render(<Blog posts={ getTestPosts() } />)
    expect(container).toMatchSnapshot()
  })
})
