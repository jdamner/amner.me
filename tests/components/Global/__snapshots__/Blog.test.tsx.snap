// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders correctly 1`] = `
<div>
  <section
    class="bg-slate-200 py-5"
    id="blog"
  >
    <div
      class="container mx-auto px-3 md:px-0"
    >
      <div
        class="flex flex-col md:flex-row md:items-end mb-5"
      >
        <h2
          class="text-lg font-sans font-semibold tracking-tight text-left font-black uppercase text-xl md:text-6xl break-words basis-1/3"
        >
          Projects and Writing
        </h2>
        <div
          class="w-full text-left uppercase text-2xl font-black"
        >
          <div
            class="mb-2"
          >
            My blog
          </div>
          <div
            class="bg-black h-2 w-full"
            style="max-width: 0px;"
          />
        </div>
      </div>
      <div
        class="columns-1 xs:columns-2 md:columns-3 lg:columns-4 gap-3"
      >
        <div
          class="break-inside-avoid-column"
          style="opacity: 0;"
        >
          <a
            class="group mb-3 block rounded-md overflow-hidden w-full bg-white relative"
            href="test-post"
          >
            <img
              alt="Test Post"
              class="w-full brightness-90 group-hover:brightness-105 transition"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <h2
              class="bg-gradient-to-r from-teal-900 to-emerald-600 group-hover:from-emerald-700 group-hover:to-teal-500 shadow-slate-900 px-4 text-lg font-sans font-semibold tracking-tight leading-loose text-center text-white normal-case"
            >
              Test Post
            </h2>
          </a>
        </div>
        <div
          class="break-inside-avoid-column"
          style="opacity: 0;"
        >
          <a
            class="group mb-3 block rounded-md overflow-hidden w-full bg-white relative"
            href="test-post"
          >
            <img
              alt="Test Post"
              class="w-full brightness-90 group-hover:brightness-105 transition"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <h2
              class="bg-gradient-to-r from-teal-900 to-emerald-600 group-hover:from-emerald-700 group-hover:to-teal-500 shadow-slate-900 px-4 text-lg font-sans font-semibold tracking-tight leading-loose text-center text-white normal-case"
            >
              Test Post
            </h2>
          </a>
        </div>
        <div
          class="break-inside-avoid-column"
          style="opacity: 0;"
        >
          <a
            class="group mb-3 block rounded-md overflow-hidden w-full bg-white relative"
            href="test-post"
          >
            <img
              alt="Test Post"
              class="w-full brightness-90 group-hover:brightness-105 transition"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <h2
              class="bg-gradient-to-r from-teal-900 to-emerald-600 group-hover:from-emerald-700 group-hover:to-teal-500 shadow-slate-900 px-4 text-lg font-sans font-semibold tracking-tight leading-loose text-center text-white normal-case"
            >
              Test Post
            </h2>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
